src/components/i-calendar.vue

```diff
@@ -0,0 +1,537 @@
+            </div>
+            <div class="content-container">
+              <slot
+                name="content"
+                :is-selected="isSelectedDate(date.date)"
+                :date="date.date"
+                :index="index"
+              />
+            </div>
+          </button>
+        </div>
+        <div
+          v-for="date in nextPicker"
+          :key="`${date.date()}${date.month()}${date.year()}-next`"
+          class="each-date next-date"
+        >
+          <div class="hidden-date">
+            {{ date.date() }}
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import dayjs from 'dayjs';
+
+import IcAngleCircle from '@/icons/ic-angle-circle.vue';
+
+export default {
+  name: 'ICalendar',
+  components: {
+    IcAngleCircle,
+  },
+  props: {
+    value: {
+      type: [Date, Array],
+      default: () => null,
+    },
+    initialDate: {
+      type: Date,
+      default: undefined,
+    },
+    disabledPreviousMonth: Boolean,
+    disabledNextMonth: Boolean,
+    isThereNewUpdate: Boolean,
+  },
+  data() {
+    return {
+      activeDate: undefined,
+      activeMonth: null,
+      activeYear: null,
+      selectedDate: [],
+      activeMonthPicker: undefined,
+      activeYearPicker: undefined,
+      showMonthYearTable: false,
+    };
+  },
+  computed: {
+    listDays() {
+      return Array.from(Array(7), (v, i) =>
+        dayjs()
+          .day(i + 0)
+          .format('ddd'),
+      );
+    },
+    listMonth() {
+      const x = Array.from(Array(12), (v, i) => ({
+        monthDisplay: dayjs(this.activeYearPicker)
+          .month(i + 0)
+          .format('MMM'),
+        monthValue: dayjs(this.activeYearPicker)
+          .month(i + 0)
+          .startOf('month'),
+      }));
+
+      return x;
+    },
+    activeDateString() {
+      return this.activeDate ? this.activeDate.toString() : undefined;
+    },
+    activeYearPickerString() {
+      return dayjs(this.activeYearPicker.toString()).format('YYYY');
+    },
+    previousPicker() {
+      const display = [];
+
+      const previous = dayjs(this.activeDateString).date(0);
+      const current = dayjs(this.activeDateString).date(0);
+      for (let i = 0; i <= current.day(); i += 1) {
+        display.push(previous.subtract(i, 'day'));
+      }
+      return display.sort((a, b) => a.day() - b.day());
+    },
+    currentPicker() {
+      return Array.from(Array(dayjs(this.activeDateString).daysInMonth()), (v, i) => {
+        const date = dayjs(this.activeDateString).date(++i);
+        return { date };
+      });
+    },
+    nextPicker() {
+      const display = [];
+      const previous = this.previousPicker.length;
+      const current = dayjs(this.activeDateString).daysInMonth();
+      for (let i = 1; i <= 42 - (previous + current); i += 1) {
+        display.push(dayjs(this.activeDateString).date(i).add(1, 'month'));
+      }
+      return display;
+    },
```

Comment:
consider to remove nextPicker, since it doesn't needed for this component

Reply:
ok